### PR TITLE
[14.0][ADD] pos_global_discount_in_line

### PR DESCRIPTION
**Pull request by ilyasProgrammer**
Include POS order discount in lines instead of adding discount product.

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
functional review ok 

from description: Include POS order discount in line instead of adding discount product.

---
this module works exactly like pos_discount, but instead of adding a "discount product", the value of discount is added in PoS order lines

File: pos_global_discount_in_line/models/pos_config.py
Comment (ivantodorovich) [inline pos_global_discount_in_line/models/pos_config.py]:
```suggestion
        string="Add Discount in Line"
```

it's already `False` by default

**Review by ivantodorovich — APPROVED** [review]

File: pos_global_discount_in_line/models/pos_config.py
Comment (ilyasProgrammer) [reply pos_global_discount_in_line/models/pos_config.py]: It is null by default. At least from the point of database there is difference between null and false.

Comment (ivantodorovich) [reply pos_global_discount_in_line/models/pos_config.py]: For boolean fields, the ORM will convert all falsy values to `False`: https://github.com/odoo/odoo/blob/939ffdfadd82789d8ad17245f48814270755b042/odoo/fields.py#L1210

No default value is the same as `default=False`, so it's just redundant.
In any case, I'm not blocking on this. Just a suggestion

**Comment by francesco-ooops** [comment]
@OCA/pos-maintainers can we merge this?

**Comment by ivantodorovich** [comment]
I believe we do.
Could you squash your commits before, please? 🙏🏻

**Comment by francesco-ooops** [comment]
@ilyasProgrammer actually now that I check again order total is not updated correctly after applying discount ([video](https://recordit.co/9W120YcaJD))

**Comment by francesco-ooops** [comment]
@ilyasProgrammer ok squash again 😎

**Comment by ilyasProgrammer** [comment]
@ivantodorovich  Squashed. + upd to remove rounding done by pos_discount module in onClick().

**Review by francesco-ooops — APPROVED** [review]
Fix works well (functional review)

Decimals for discount value were not passed correctly to PoS order lines

@ivantodorovich can you take a look? thanks!

**Comment by ivantodorovich** [comment]
LG 👍🏻 
Could you squash again, however? I still see two commits 😅

**Comment by francesco-ooops** [comment]
@ivantodorovich should be good?

**Comment by ivantodorovich** [comment]
indeed

/ocabot merge nobump

**Pull request merged**
